flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionTest.java

```diff
@@ -478,22 +489,25 @@ public void testEagerSchedulingFailureReturnsSlot() throws Exception {
 						slotRequestId);
 					slotProvider.complete(slotRequestId, singleLogicalSlot);
 				},
-				executorService);
-
-			final CompletableFuture<Void> schedulingFuture = execution.scheduleForExecution(
-				slotProvider,
-				false,
-				LocationPreferenceConstraint.ANY,
-				Collections.emptySet());
-
-			try {
-				schedulingFuture.get();
-				// cancel the execution in case we could schedule the execution
-				execution.cancel();
-			} catch (ExecutionException ignored) {
-			}
-
-			assertThat(returnedSlotFuture.get(), is(equalTo(slotRequestIdFuture.get())));
+				executorService).thenRunAsync(() -> {
+					try {
+						final CompletableFuture<Void> schedulingFuture = execution.scheduleForExecution(
+							slotProvider,
+							false,
+							LocationPreferenceConstraint.ANY,
+							Collections.emptySet());
+
+						try {
+							schedulingFuture.get();
+							// cancel the execution in case we could schedule the execution
+							execution.cancel();
+						} catch (ExecutionException ignored) {
+						}
+
+						assertThat(returnedSlotFuture.get(), is(equalTo(slotRequestIdFuture.get())));
+					} catch (Exception ex) {
+					}
+			}, testMainThreadUtil.getMainThreadExecutor());
```

Comment:
I think it is not that easy because there is curently no way to complete the future after scheduling took the failing branch without using another thread.